media/sse/src/main/java/org/glassfish/jersey/media/sse/internal/JerseySseBroadcaster.java

```diff
@@ -102,10 +103,9 @@ public void onClose(Consumer<SseEventSink> onClose) {
         if (event == null) {
             throw new IllegalArgumentException(LocalizationMessages.PARAM_NULL("event"));
         }
-        publish(event);
 
         // TODO JAX-RS 2.1
```

Comment:
Should this TODO be removed?

Reply:
@Cousjava We do not feel strongly about the TODO. Ideally, the `CompletableFuture#isDone()` would return true AFTER the messages are broadcasted (which possibly was the Spec intention), but it does not happen now and it is hard to accomplish using the JSR166 SubmissionPublisher. So we tend to leave it there as a mark of what have been made better.  The PR is mainly not to return null. Do you feel the TODO should be removed?